packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -376,5 +427,55 @@ export default createRule<Options, MessageId>({
       const type = getConstrainedTypeAtLocation(services, targetNode);
       return tsutils.isTypeFlagSet(type, ts.TypeFlags.VoidLike);
     }
+
+    function hasValidReturnType(
+      services: ParserServicesWithTypeInformation,
```

Comment:
[Refactor] (optional) `hasValidReturnType` function is declared in block where `context` is visible. What do you think about calling `const services = getParserServices(context);` inside `hasValidReturnType` (like `canFix` function above does) instead of passing `services` as an argument?

Reply:
I too believe that using context is better than passing a service.